chill-java/src/main/java/com/twitter/chill/java/BitSetSerializer.java

```diff
@@ -92,15 +93,14 @@ public BitSet read(Kryo kryo, Input input, Class<BitSet> bitSetClass) {
         // call a private constructor: (the BitSet.valueOf() cTor is only available from Java 1.7)
         try {
             ret = bitSetConstructor.newInstance(target);
-        } catch (ReflectiveOperationException e) {
-            throw new KryoException("Unable to call BitSet(long[]) constructor", e);
-        }
-        try {
             recalculateWordsInUseMethod.invoke(ret);
-        } catch (ReflectiveOperationException e) {
-            throw new KryoException("Unable to call BitSet.recalculateWordsInUse() method", e);
+        } catch (InvocationTargetException e) {
+            throw new KryoException("Unable to call BitSet(long[]) constructor", e);
+        } catch (IllegalAccessException e) {
+            throw new KryoException("Unable to call BitSet(long[]) constructor", e);
+        } catch (InstantiationException e) {
```

Comment:
can you add one more clause for `RuntimeException` to cover `IllegalArgumentException` and friends, and update the Strings in these exceptions